services/blockchain-indexer/shared/dataService/business/transactionsEstimateFees.js

```diff
@@ -262,6 +264,41 @@ const estimateTransactionFees = async params => {
 		throw new ValidationException('Incorrect \'recipientAddress\' specified in transaction params.');
 	}
 
+	if (blsKey && !regex.BLS_KEY.test(blsKey)) {
+		throw new ValidationException('Incorrect \'blsKey\' specified in transaction params.');
+	}
+
+	if (proofOfPossession && !regex.PROOF_OF_POSSESSION.test(proofOfPossession)) {
+		throw new ValidationException('Incorrect \'proofOfPossession\' specified in transaction params.');
+	}
+
+	if (generatorKey && !regex.PUBLIC_KEY.test(generatorKey)) {
+		throw new ValidationException('Incorrect \'generatorKey\' specified in transaction params.');
+	}
+
+	if (receivingChainID && !regex.CHAIN_ID.test(receivingChainID)) {
+		throw new ValidationException('Incorrect \'receivingChainID\' specified in transaction params.');
+	}
+
+	if (sendingChainID && !regex.CHAIN_ID.test(sendingChainID)) {
+		throw new ValidationException('Incorrect \'sendingChainID\' specified in transaction params.');
+	}
+
+	if (messageFeeTokenID && !regex.TOKEN_ID.test(messageFeeTokenID)) {
+		throw new ValidationException('Incorrect \'messageFeeTokenID\' specified in transaction params.');
+	}
```

Comment:
Would it have better code if we integrate the checks in [gateway file ](https://github.com/LiskHQ/lisk-service/blob/51b074adc43117d70d514fc622f4ce7b8d09c402/services/blockchain-connector/methods/encoder.js#L26-L36)and indexer methods [file](https://github.com/LiskHQ/lisk-service/blob/6d8b0c44a81777401d38a84d3719927402bcab68/services/blockchain-indexer/methods/dataService/transactions.js#L79C36-L79C50) instead? we won't need the `validateTransactionParams` function in that case.

[Example](https://github.com/LiskHQ/lisk-service/blob/51b074adc43117d70d514fc622f4ce7b8d09c402/services/blockchain-connector/methods/encoder.js#L26-L36)

```suggestion
```

Reply:
Updated the code to use `validator.validate` to validate all tx params instead.